### PR TITLE
Migrate GiftedChat class component to functional component and hooks

### PR DESCRIPTION
**Pull request by amerikan**
Modernizes the `GiftedChat` component, which currently is written using Class component, to now be written using functional components with hooks.

## Goals
- modernize `GiftedChat.tsx` component
- no breaking changes
- minimize code changes to focus on making it work with function component and hooks, rather than optimizing or doing clean ups 

## Future Goals
- remove redundant code
- separate state
- other improvements

### CODE REVIEW

File: src/GiftedChat.tsx [hunk 1]
```diff
@@ -232,360 +232,233 @@ export interface GiftedChatState<TMessage extends IMessage = IMessage> {
...
-    onInputTextChanged: null,
-    maxInputLength: null,
-    forceGetKeyboardHeight: false,
-    inverted: true,
-    extraData: null,
-    minComposerHeight: MIN_COMPOSER_HEIGHT,
-    maxComposerHeight: MAX_COMPOSER_HEIGHT,
-  }
-
-  static propTypes = {
```
Comment (amerikan) [inline src/GiftedChat.tsx:308]: I removed all propTypes in favor of relying on Typescripts type checking instead.

File: src/GiftedChat.tsx [hunk 2]
```diff
@@ -232,360 +232,233 @@ export interface GiftedChatState<TMessage extends IMessage = IMessage> {
...
     typingDisabled: false,
     text: undefined,
     messages: undefined,
-  }
+  })
 
-  constructor(props: GiftedChatProps<TMessage>) {
-    super(props)
-
-    this.invertibleScrollViewProps = {
```
Comment (amerikan) [inline src/GiftedChat.tsx:422]: Moved this to directly where it's being used. https://github.com/FaridSafi/react-native-gifted-chat/pull/2372/files#diff-b11e3f8b30e19689a33f545b3567228eb41fd9cfbbad474b1b21e7b63eaa2d57R567

**Comment by amerikan** [comment]
@Johan-dutoit perfect! Let me know if you have any questions. I'll be available to help!

**Comment by Eigilak** [comment]
Looks good to me!

**Comment by Johan-dutoit** [comment]
Merged unintentionally 😅 
Mostly looking good, a few type issues and unused functions.  

Let's move forward instead of revert @amerikan

**Pull request merged**